packages/samples/headless-react/src/components/triggers/execute-trigger.tsx

```diff
@@ -0,0 +1,27 @@
+import {buildExecuteTrigger, SearchEngine} from '@coveo/headless';
+
+/**
+ * This sample create a headless execute trigger controller instance
+ * and shows how to utilize the controller with the user defined functions.
+ *
+ * @param engine - a headless search engine instance
+ * @returns An unsubscribe function
+ */
+export function bindExecuteTrigger(engine: SearchEngine) {
```

Comment:
I decided to keep the class component but got rid of the functional component and replaced it with this function instead.